app/views/unattended/provisioning_templates/snippet/disk_enc_clevis_tang.erb

```diff
@@ -27,52 +27,66 @@ description: |
 <% if (@host.operatingsystem.family == 'Redhat' || @host.operatingsystem.name == 'Ubuntu') && tang_server_list.present? -%>
 
 cat > /tmp/rootdir-luks-device.sh << "EOF"
-#!/bin/sh
+#!/bin/bash
 #
-# Author Jan Löser <loeser@atix.de>
-# Published under the GNU Public Licence 3
+# SPDX-FileCopyrightText: 2024 Jan Löser <loeser@atix.de>
+# SPDX-FileCopyrightText: 2024 Martin Spiessl <spiessl@atix.de>
+# SPDX-License-Identifier: GPL-3.0-or-later
 #
-# This scripts tries to find the 1st LUKS device for / (root directory).
+# This script tries to find all LUKS devices mounted in /etc/crypttab.
 #
 set -o pipefail
 
-rootdev=$(df / --output=source | tail -n1)
-targetdev=$(readlink -f $rootdev)
-slavedev=$targetdev
-
-while : ; do
-    /sbin/cryptsetup luksDump $slavedev &>/dev/null && echo $slavedev && exit 0
+mapfile -t maybe_luks_devices < <(
+    awk '{print $1}' /etc/crypttab | \
```

Comment:
hopefully everyone has awk installed.

Reply:
never though about it, but true! This made me think how you could trust any utility to be present really, and I found out that `awk` and `xargs` are at least part of the POSIX standard's mandatory utilities. People rarely claim their OS is (fully) POSIX compliant, but at least that gives some justification for why we expect it.

I can get rid of this `awk` call if you think it is desirable, but rather would not go down that road with other precious utilities like `xargs` :sweat_smile:

Reply:
@sbernhard not sure if this is resolved or you argue for removing `awk` calls (and then what about `xargs`?). I do not mind refactoring those calls away if you thing this is something that should be done, but if you can live without it the code would stay a lot more readable.

Reply:
I'm ok. I guess, awk should be installed.